hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/scm/ReconScmTask.java

```diff
@@ -0,0 +1,103 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.ozone.recon.scm;
+
+import org.hadoop.ozone.recon.schema.tables.daos.ReconTaskStatusDao;
+import org.hadoop.ozone.recon.schema.tables.pojos.ReconTaskStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.inject.Inject;
+
+/**
+ * Any background task that keeps SCM's metadata up to date.
+ */
+public abstract class ReconScmTask {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ReconScmTask.class);
+  private Thread taskThread;
+  private ReconTaskStatusDao reconTaskStatusDao;
+  private volatile boolean running;
+
+  @Inject
+  public ReconScmTask(ReconTaskStatusDao reconTaskStatusDao) {
+    this.reconTaskStatusDao = reconTaskStatusDao;
+  }
+
+  public void register() {
+    String taskName = getClass().getSimpleName();
+    ReconTaskStatus reconTaskStatusRecord = new ReconTaskStatus(
+        taskName, 0L, 0L);
+    if (!reconTaskStatusDao.existsById(taskName)) {
+      reconTaskStatusDao.insert(reconTaskStatusRecord);
```

Comment:
Do we need to always create `ReconTaskStatus`, or only if it is being inserted?

Reply:
Good catch, will fix it.